### PR TITLE
Add dscinitializations resource to cluster-role.yaml and handle missing dsci status in ServingRuntimeList.cy.ts

### PR DESCRIPTION
**Pull request by lucferbux**
Closes https://issues.redhat.com/browse/RHOAIENG-6578 

## Description

Add permissions to fetch dsci by the dashboard serviceaccount and fixes a flaw in the area enablement util method that returned true if the dsci was missing.

## How Has This Been Tested?

1. Run the changes in a cluster with dsci but no permissions
2. Use `npm run start:dev:ext` to get the deployed dashboard serviceaccount capabilities
3. Check that the kserve auth feature should be disabled
4. Now add the permissions to the cluster (see bellow)
5. Check that the feature should be enabled

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [X] The developer has tested their solution on a cluster by using the image produced by the PR to `main`
**Tested with npm run start:dev:ext which sholud mimic this same behavior**

### CODE REVIEW

File: frontend/src/concepts/areas/utils.ts [hunk 1]
```diff
@@ -72,18 +72,18 @@ export const isAreaAvailable = (
...
-            ...acc,
-            [capability]: dsciStatus.conditions.some(
-              (c) => c.type === capability && c.status === 'True',
-            ),
-          }),
-          {},
-        )
-      : null;
+  // If DSCI is not available, we shouldn't assume it has met the required capabilities
+  const requiredCapabilitiesState = requiredCapabilities
```
Comment (andrewballantyne) [inline frontend/src/concepts/areas/utils.ts:76]: Why the change 🤔 We have this for DSC

Comment (lucferbux) [reply frontend/src/concepts/areas/utils.ts:76]: This is part of the issue that i discovered, if DSCI returns a 404 (it's happening in a cluster cause it lacks the permission), it will later return true as if the condition is fulfilled.

Comment (lucferbux) [reply frontend/src/concepts/areas/utils.ts:76]: We shouldn't get that state as dsci should be present every time, but we met this specific edge case @andrewballantyne

Comment (andrewballantyne) [reply frontend/src/concepts/areas/utils.ts:76]: This is true of everything -- including the DSC -- this is not a good reason to change the logic on the frontend.

Comment (andrewballantyne) [reply frontend/src/concepts/areas/utils.ts:76]: There is a reason we have the template "test on cluster" checkbox. Because of weird permission things like this.

Comment (lucferbux) [reply frontend/src/concepts/areas/utils.ts:76]: > There is a reason we have the template "test on cluster" checkbox. Because of weird permission things like this.
Yes, and that was on me, I missed that cause when we shipped this the operator side wasn't even completed, so I didn't have the ability to test it on cluster, just mocking my own responses, and that's on me.

On the other hand, I've tested it right now and it seems that we can get the same result with `npm run start:dev:ext` which will give us much more flexibility to test permissions on clusters

Comment (lucferbux) [reply frontend/src/concepts/areas/utils.ts:76]: I'm gonna remove the logic change, keep the tests and keep the permission.

**Review by andrewballantyne — APPROVED** [review]
Lucas tested with connecting to the Dashboard Pod and overriding permissions on cluster -- I'm approving code.
